project/ui/HUD.cpp

```diff
@@ -0,0 +1,102 @@
+#include "HUD.h"
+#include "imgui/imgui.h"
+#include "graphics/d3dresource/GraphicsResource.h"
+#include "LogUIElement.h"
+#include "TurnPhaseUIElement.h"
+
+namespace
+{
+    constexpr auto HUD_WINDOW_FLAGS = ImGuiWindowFlags_MenuBar | ImGuiWindowFlags_NoCollapse | ImGuiWindowFlags_NoTitleBar | ImGuiWindowFlags_NoResize;
```

Comment:
This line is getting kind of wide. I think vertically aligning theses would look nicer.
ImGuiWindowFlags_MenuBar |
ImGuiWindowFlags_NoCollapse | 
ImGuiWindowFlags_NoTitleBar | 
ImGuiWindowFlags_NoResize;